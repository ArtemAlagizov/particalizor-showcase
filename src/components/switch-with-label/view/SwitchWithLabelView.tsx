import React from "react";
import { Styled } from "../styled";
import { ISwitchWithLabelView } from "./ISwitchWithLabelView";
import { Switch } from "antd";

const SwitchWithLabelView: React.FC<ISwitchWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.SwitchWithLabelContainer>
      <Styled.SwitchLabel>{label}</Styled.SwitchLabel>
      <Styled.SwitchContainer>
        <Switch defaultChecked={defaultValue} onChange={onChange} />
      </Styled.SwitchContainer>
    </Styled.SwitchWithLabelContainer>
  );
};

export default SwitchWithLabelView;
