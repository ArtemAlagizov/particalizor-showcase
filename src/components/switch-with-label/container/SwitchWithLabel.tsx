import React from "react";
import SwitchWithLabelView from "../view/SwitchWithLabelView";
import { ISwitchWithLabel } from "./ISwitchWithLabel";

export const SwitchWithLabel: React.FC<ISwitchWithLabel> = ({
  onValueChange,
  propertyName,
  defaultValue,
  label
}) => {
  const onChange = (value: boolean) => onValueChange(value, propertyName);

  return (
    <SwitchWithLabelView
      label={label}
      defaultValue={defaultValue}
      onChange={onChange}
    />
  );
};

export default SwitchWithLabel;
