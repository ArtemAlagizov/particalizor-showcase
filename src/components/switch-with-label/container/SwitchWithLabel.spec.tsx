import React from "react";
import { mount, shallow } from "enzyme";
import SwitchWithLabel from "./SwitchWithLabel";
import SwitchWithLabelView from "../view/SwitchWithLabelView";

describe("Given a SwitchWithLabel", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <SwitchWithLabel
          onValueChange={jest.fn()}
          label={""}
          propertyName={""}
          defaultValue={false}
        />
      );

      expect(component).toMatchSnapshot();
    });
    it("should render MovingPictureView", () => {
      const component = mount(
        <SwitchWithLabel
          onValueChange={jest.fn()}
          label={""}
          propertyName={""}
          defaultValue={false}
        />
      );

      expect(component.find(SwitchWithLabelView).length).toBe(1);
    });
  });
});
