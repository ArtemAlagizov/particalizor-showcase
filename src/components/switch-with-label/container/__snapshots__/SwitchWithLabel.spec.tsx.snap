// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SwitchWithLabel when rendering should match snapshot 1`] = `
<SwitchWithLabelView
  defaultValue={false}
  label=""
  onChange={[Function]}
/>
`;
